### PR TITLE
Add setting to set a different primary color for dark themes

### PR DESCRIPTION
**Pull request by artonge**
The primary color can be in-adapted to dark theme, this new settings allow to choose a different one.
If the color for dark theme is not set, it will default to the primary one.

Screencast from 19-05-2022 15:57:37

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: apps/theming/lib/ThemingDefaults.php [hunk 1]
```diff
@@ -228,8 +228,22 @@ public function getShortFooter() {
 	 *
 	 * @return string
 	 */
-	public function getColorPrimary() {
-		$color = $this->config->getAppValue('theming', 'color', $this->color);
+	public function getColorPrimary(bool $dark = false) {
+		$color = $this->config->getAppValue(
+			'theming',
+			$dark ? 'color_dark_theme' : 'color',
+			null
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/theming/lib/ThemingDefaults.php:235]: Why not, but I would love a rational :)

[the remark this answers is not in the document]
Comment (artonge) [reply apps/theming/lib/ThemingDefaults.php]: Ok, psalm was not happy :sweat_smile:

**Review by Pytal — APPROVED** [review]

**Review by skjnldsv — CHANGES_REQUESTED** [review]
Blocking

**Comment by skjnldsv** [comment]
see for fix https://github.com/nextcloud/server/pull/32509

**Pull request closed without merging**